Data aggregation processors added; Issue #3: Stdin support added; Issue: #5 JSON config support added

b0s-works opened the pull request:
Added dynamic type data aggregation summarization methods: avg, sum and countUnique
Structure separated by two part: stdin receiver and files reader blocks
Work isn't done, but needs a review, comments and observations to guide me to the right developing path

Best Regards, Ivan

Commit:
Some changes

Commit:
Added dynamic type data aggregation summarization methods: avg, sum and countUnique
Structure separated by two part: stdin receiver and files reader blocks
Work isn't done, but needs a review, comments and observations to guide me to the right developing path
Program became runable, but work isn't done.

Status Early alpha

oleg-raev commented on main.go:
```diff
@@ -1,13 +1,17 @@
+	"github.com/sirupsen/logrus"
 )
 
 /*
```
Remove comment, specified by CSV file

oleg-raev commented on main.go:
```diff
@@ -35,41 +65,85 @@ func main() {
-	data := strings.Split(row, ";")
-	if len(data) < 1 || len(data[0]) < 13 {
-		return
+func main() {
```
Main function should be on top

Commit:
Removed Comment, dependent from CSV
Main function on top now
ValueSaver interface renamed to DataProcessor
summarizer.go.Storage type renamed to Result
Summarizer processors structures( avg.go, count-unique.go, sum.go ) renamed
File reader and StdIn functions moved to input.go
Flag package used to determine using of «-stdin» command line parameter to change source to stdin
Itoa and FormatInt used instead of Sprintf to convert int/int64 type numbers to string
ProcessRow function now part of Storage type, so this func became shorter and more flexible
